### DISCUSSION TITLE
Checking if user is logged in without redirection

### DISCUSSION THREAD

**Discussion by praagyajoshi** [meza/auth0-remix-server#79] [Q&A]
I have a use case where I need to check whether a user is logged in or not — but being logged in is not a requirement.  
From what I understand, this is not possible currently due to the fact that `getUser()` currently throws a `redirect()` if:  
- credentials aren't available https://github.com/meza/auth0-remix-server/blob/main/src/index.ts#L201
- credentials refresh didn't work https://github.com/meza/auth0-remix-server/blob/main/src/index.ts#L224

How should I go about implementing this behaviour? I'm happy to submit a PR for this added functionality.

I was thinking passing an option to `getUser()` called `redirectOnFailure` and:
- throwing the redirect if `redirectOnFailure` is `true`
- otherwise, returning `null`

Thank you! 🙏

**Comment by meza** [comment]
Hey @praagyajoshi,

you could simply wrap the getUser() in a try/catch. Try would be the logged-in path and the catch would be the logged out path
